pkg/collector/corechecks/windows_event_log/config_helpers.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+package evtlog
+
+import (
+	"fmt"
+	"regexp"
+
+	agentEvent "github.com/DataDog/datadog-agent/pkg/metrics/event"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/eventlog/api"
+)
+
+func compileRegexPatterns(patterns []string) ([]*regexp.Regexp, error) {
+	var err error
+	res := make([]*regexp.Regexp, len(patterns))
+	for i, pattern := range patterns {
+		res[i], err = regexp.Compile(pattern)
+		if err != nil {
+			return nil, fmt.Errorf("error compiling regex pattern '%s': %w", pattern, err)
+		}
+	}
+	return res, nil
+}
+
+func serverIsLocal(server util.Optional[string]) bool {
+	val, isSet := server.Get()
+	return !isSet ||
+		len(val) == 0 ||
+		val == "localhost" ||
+		val == "127.0.0.1"
+}
+
+func evtRPCFlagsFromString(flags string) (uint, error) {
+	// NOTE: Keep this in sync with config spec `auth_type`
+	switch flags {
+	case "default":
+		return evtapi.EvtRpcLoginAuthDefault, nil
+	case "negotiate":
+		return evtapi.EvtRpcLoginAuthNegotiate, nil
+	case "kerberos":
+		return evtapi.EvtRpcLoginAuthKerberos, nil
+	case "ntlm":
+		return evtapi.EvtRpcLoginAuthNTLM, nil
+	default:
+		return 0, fmt.Errorf("invalid RPC auth type: '%s', must be one of default, negotiate, kerberos, ntlm", flags)
+	}
+}
+
+func evtRPCFlagsFromOption(authType util.Optional[string]) (uint, error) {
+	val, isSet := authType.Get()
+	if !isSet {
+		return 0, fmt.Errorf("option is not set")
+	}
+	return evtRPCFlagsFromString(val)
+}
+
+func isaffirmative(o util.Optional[bool]) bool {
```

Comment:
Would it make sense to support the same range of values as the python check for backwards compatibility?

https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/config.py#L12

If not, and all you care if option was set to true, then `Optional` is redundant. Missing fields already automatically deserialize to a zero value (false for bool).

Reply:
> If not, and all you care if option was set to true, then Optional is redundant.

We need optional because we need to check the "not set" state to determine how to propagate the `init_config` and the default values to the final config used by the check. it is true that after the config is done being processed that we don't need Optional anymore, but we'd have to create another struct type to hold the final values and copy them over, I didn't think it was worth it.

> Would it make sense to support the same range of values as the python check for backwards compatibility?

go yaml provides support for all of those values to `bool` except for 1 and 0 :(

I'll see if I can add support for that. I think it'll require 👇
> but we'd have to create another struct type to hold the final values and copy them over,